mod_test.go

```diff
@@ -0,0 +1,61 @@
+package mod
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+)
+
+func TestGetModFile(t *testing.T) {
+	dir := t.TempDir()
+	err := os.WriteFile(filepath.Join(dir, "go.mod"), []byte(`
+module example.com/foo
+
+require example.com/whatever v1.2.3
+
+tool example.com/whatever/cmd/thing
+
+go 1.23.4
+`), 0o644)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	mod, err := GetModFile(dir)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if mod.Module.Mod.Path != "example.com/foo" {
+		t.Fatalf("unexpected module path: %v", mod.Module.Mod.Path)
+	}
+}
+
+func TestGetModFileErrors(t *testing.T) {
+	t.Run("DoesNotExist", func(t *testing.T) {
+		_, err := GetModFile(t.TempDir())
+		if err == nil {
+			t.Fatal("expected error")
+		}
+		if !strings.Contains(err.Error(), "could not open go.mod") {
+			t.Fatalf("unexpected error: %v", err)
+		}
```

Comment:
Let's not test on strings as that could change without notice. Just checking `if err == nil` is good enough unless we propagate the os.NotExist err that we can check against.